app/controllers/comments_controller.rb

```diff
@@ -22,15 +22,17 @@ def index
         Organization.find_by(slug: params[:username]) ||
         not_found
       @commentable = @root_comment&.commentable ||
-        @user.articles.find_by(slug: params[:slug]) ||
-        not_found
+        @user.articles.find_by(slug: params[:slug]) || nil
       @article = @commentable
-      not_found unless @commentable.published
+      if @commentable
+        not_found unless @commentable.published
+      end
+      render :deleted_article_comment unless @article
```

Comment:
AFAIK, commentable can also not be an article (podcast episode have comments too) so I guess this has to be generalized as well

Reply:
@rhymes  I just pushed up a commit to show comments for deleted podcasts as well. 

Also, I noticed `@on_comments_page = true` on the comments#index but I am not seeing `@on_comments_page` used anywhere. Doesn't look like it's needed.